meson.build

```diff
@@ -0,0 +1,3419 @@
+           link_with : [libnetworkd_core,
+                        libfirewall,
+                        libsystemd_network,
+                        libudev_internal,
+                        libshared],
+           install : true,
+           install_dir : rootlibexecdir)
+
+executable('systemd-networkd-wait-online',
+           systemd_networkd_wait_online_sources,
+           include_directories : includes,
+           link_with : [libnetworkd_core,
+                        libshared],
+           install : true,
+           install_dir : rootlibexecdir)
+
+executable('networkctl',
+           networkctl_sources,
+           include_directories : includes,
+           link_with : [libsystemd_network,
+                        libshared],
+           install : true)
+
+############################################################
+
+path = run_command('sh', ['-c', 'echo "$PATH"']).stdout()
+test_env = environment()
+test_env.set('SYSTEMD_KBD_MODEL_MAP', kbd_model_map)
+test_env.set('SYSTEMD_LANGUAGE_FALLBACK_MAP', language_fallback_map)
+test_env.set('PATH', path)
+test_env.prepend('PATH', meson.build_root())
+
+test_device_nodes = executable(
+    'test-device-nodes',
+    'src/test/test-device-nodes.c',
+    include_directories : includes,
+    link_with : [libshared])
+test('test-device-nodes', test_device_nodes,
+     env : test_env)
+
+test_engine = executable(
+    'test-engine',
+    'src/test/test-engine.c',
+    include_directories : includes,
+    link_with : [libcore],
+    dependencies : [librt,
+                    libseccomp,
+                    libselinux,
+                    libmount])
+test('test-engine', test_engine,
+     env : test_env)
+
+test_job_type = executable(
+    'test-job-type',
+    'src/test/test-job-type.c',
+    include_directories : includes,
+    link_with : [libcore],
+    dependencies : [librt,
+                    libseccomp,
+                    libselinux,
+                    libmount])
+test('test-job-type', test_job_type,
+     env : test_env)
+
+test_ns = executable(
+    'test-ns',
+    'src/test/test-ns.c',
+    include_directories : includes,
+    link_with : [libcore],
+    dependencies : [librt,
+                    libseccomp,
+                    libselinux,
+                    libmount])
+
+test_loopback = executable(
+    'test-loopback',
+    'src/test/test-loopback.c',
+    include_directories : includes,
+    link_with : [libcore],
+    dependencies : [librt,
+                    libseccomp,
+                    libselinux,
+                    libmount])
+test('test-loopback', test_loopback,
+     env : test_env)
+
+test_hostname = executable(
+    'test-hostname',
+    'src/test/test-hostname.c',
+    include_directories : includes,
+    link_with : [libcore],
+    dependencies : [librt,
+                    libseccomp,
+                    libselinux,
+                    libmount])
+if want_tests == 'unsafe'
+  test('test-hostname', test_hostname,
+       env : test_env)
+endif
+
+
```

Comment:
Note that `Makefile.am` contains
```sh
# Link statically because this test uses BUS_ERROR_MAP_ELF_REGISTER
test_bus_error_LDADD = \
        libshared.la
```
This is why we got `Symbol __stop_BUS_ERROR_MAP' causes overflow in R_X86_64_PC32 relocation`.